main.go

```diff
@@ -0,0 +1,393 @@
+		})
+		if err != nil {
+			glog.Errorf("Error creating CRM credentials for  user %s,  %v", u.PrimaryEmail, err)
+			return []*cloudresourcemanager.Project{}, err
+		}
+		crmService, err = cloudresourcemanager.NewService(ctx, option.WithCredentials(cred))
+		if err != nil {
+			glog.Errorf("Error creating CRM Service for  user %s,  %v", u.PrimaryEmail, err)
+			return []*cloudresourcemanager.Project{}, err
+		}
+	}
+
+	projects := make([]*cloudresourcemanager.Project, 0)
+	// do not apply a list filter here; we need to audit all projects
+	//  https://pkg.go.dev/google.golang.org/api@v0.58.0/cloudresourcemanager/v1#ProjectsListCall.Filter
+	noFilter := ""
+
+	req := crmService.Projects.List().Filter(noFilter).PageSize(maxPageSize)
+	if err := req.Pages(ctx, func(page *cloudresourcemanager.ListProjectsResponse) error {
+		projects = append(projects, page.Projects...)
+		if err := limiter.Wait(ctx); err != nil {
+			glog.Errorf("Error in rate limiter for user %s %v", u.PrimaryEmail, err)
+			return err
+		}
+		if ctx.Err() != nil {
+			glog.Errorf("Error in rate limiter for user %s %v", u.PrimaryEmail, ctx.Err())
+			return ctx.Err()
+		}
+		return nil
+	}); err != nil {
+		glog.Errorf("Error iterating visible projects for user %s %v", u.PrimaryEmail, err)
+		return []*cloudresourcemanager.Project{}, err
+	}
+
+	for _, p := range projects {
+		glog.V(50).Infof("             User %s has Project visibility to %s", u.PrimaryEmail, p.ProjectId)
+		if _, ok := allProjects[p.ProjectId]; !ok {
+			glog.V(2).Infof("             User [%s] has external project visibility to [projects/%d](%s)", u.PrimaryEmail, p.ProjectNumber, p.ProjectId)
+		}
+	}
+	return projects, nil
+}
+
+func main() {
+
+	flag.Parse()
+	defer glog.Flush()
+	ctx := context.Background()
+
+	// configure API rate limits
+	limiter = rate.NewLimiter(rate.Limit(maxRequestsPerSecond), burst)
+	if *organization == "" || *subject == "" || *cx == "" {
+		glog.Error("--organization and --cx and --subject   must be specified")
+		return
+	}
+	if (*serviceAccountFile == "" && *impersonatedServiceAccount == "") || (*serviceAccountFile != "" && *impersonatedServiceAccount != "") {
+		glog.Error("either --serviceAccountFile or --impersonatedServiceAccount must be specified")
+		return
+	}
+
+	if *serviceAccountFile != "" {
+		var err error
+		svcAccountJSONBytes, err = ioutil.ReadFile(*serviceAccountFile)
+		if err != nil {
+			glog.Fatal(err)
+		}
+	}
+
+	// now initialize the Cloud Asset API
+	//  this api is used to find the projects in an ORG
+	var assetClient *asset.Client
+	if *impersonatedServiceAccount != "" {
+		ts, err := impersonate.CredentialsTokenSource(ctx, impersonate.CredentialsConfig{
+			TargetPrincipal: *impersonatedServiceAccount,
+			Scopes:          []string{cloudresourcemanager.CloudPlatformScope},
+		})
+		if err != nil {
+			glog.Fatal(err)
+		}
+		assetClient, err = asset.NewClient(ctx, option.WithTokenSource(ts))
+		if err != nil {
+			glog.Fatal(err)
+		}
+	} else {
```

Comment:
Remove this else and move the code block to the level above the else block -- Fatal in the first block halts execution.